tasks/main.yml

```diff
@@ -48,3 +48,22 @@
     name: consul
     state: reloaded
   when: consul_service_definition.changed or consul_service_removal.changed
+
+- name: Upload a service to put consul under maintenance at shutdown
+  template:
+    src: consul-maint-before-shutdown.service.j2
+    dest: /lib/systemd/system/consul-maint-before-shutdown.service
```

Comment:
IMHO this should rather go in `/etc/systemd/system/...`. The `/lib/systemd` path is used for OS-delivered unit files.